UIA/SearchField: catch COMError when attempting to look for search fields

josephsl opened the pull request:
### Link to issue number:
Fixes #10422 

### Summary of the issue:
In Windows 10, error tone is heard when navigating through pages in Settings, as well as when opening emoji panel.

### Description of how this pull request fixes the issue:
Catch COMError thrown when attempting to add overlay class for search fields.

### Testing performed:
Tested via Windows 10 App Essentials add-on and on source code version to make sure error tone is no longer heard.

### Known issues with pull request:
None

### Change log entry:
None

### Additional context:
Although #10397 will catch this and resolve this at the UIA handler thread level, it is sometimes useful to figure out the root cause of issues, hence this pull request.

josephsl committed:
UIA/SearchField: catch COMError when attempting to look for search fields. Re #10422

Error thrown by Comtypes: COMError thrown when attempting to look for search fields. Therefore wrap this inside a try/except block. This resolves error tones heard when one navigates through Windows 10 Settings categories and subpages.

leonardder commented on source/NVDAObjects/UIA/__init__.py:
```diff
@@ -863,8 +863,11 @@ def findOverlayClasses(self,clsList):
+			if self.UIAElement.cachedAutomationID in ("SearchTextBox", "TextBox"):
+				clsList.append(SearchField)
+		except COMError:
+			pass
```
I think this should still be logged.

josephsl commented:
Hi, I see – changes under way. Thanks.

 

From: Leonard de Ruijter <notifications@github.com> 
Sent: Thursday, November 14, 2019 10:22 AM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [nvaccess/nvda] UIA/SearchField: catch COMError when attempting to look for search fields (#10492)

 

@leonardder requested changes on this pull request.

  _____  

In source/NVDAObjects/UIA/__init__.py <https://github.com/nvaccess/nvda/pull/10492#discussion_r346471747> :

> @@ -863,8 +863,11 @@ def findOverlayClasses(self,clsList):
               if isDialog:
                       clsList.append(Dialog)
               # #6241: Try detecting all possible suggestions containers and search fields scattered throughout Windows 10.
-              if self.UIAElement.cachedAutomationID in ("SearchTextBox", "TextBox"):
-                      clsList.append(SearchField)
+              try:
+                      if self.UIAElement.cachedAutomationID in ("SearchTextBox", "TextBox"):
+                              clsList.append(SearchField)
+              except COMError:
+                      pass

I think this should still be logged.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/10492?email_source=notifications&email_token=AB4AXEANZO7SQKIJPQNSRWDQTWJK7A5CNFSM4JNOOVCKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCLTYPPY#pullrequestreview-317163455> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/AB4AXECTMZ4LK4VAPL6A3QLQTWJK7ANCNFSM4JNOOVCA> .

josephsl committed:
UIA search field: log a debug log when COMError is caught. Re #10422.

Recommended by Leonard de Ruijter: record a log when COMError is thrown when locating search field objects.

josephsl committed:
UIA search field: lint
